gtsfm/common/two_view_estimation_report.py

```diff
@@ -46,3 +45,6 @@ class TwoViewEstimationReport:
     U_error_deg: Optional[float] = None
     i2Ri1: Optional[Rot3] = None
     i2Ui1: Optional[Unit3] = None
+    reproj_error_gt_model: Optional[np.ndarray] = None
```

Comment:
since we starting to get a lot of attributes here, I wonder if we should make this hierarchical. I propose that we make a report like
```python
@dataclass
class MeshReprojectionReport
    reproj_error_gt_model: float
    inlier_avg_reproj_error_gt_model: float
    outlier_avg_reproj_error_gt_model: float
```
Then 
```python
@dataclass(frozen=False)
class TwoViewEstimationReport:
    mesh_reprojection_report: Optional[MeshReprojectionReport] = None
```
What are your thoughts about that? I think it keeps things cleaner for attributes that we usually won't have.

We need docstrings explaining what each of those quantities mean here for the constructor, though.

Reply:
The reprojection error is defined for the Sampson distance metric as well.